drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -608,88 +605,62 @@ local function initialize_switch(driver, device)
+  -- is only run once, but find_child function should be set on each driver init.
   if parent_child_device then
-    -- If the device is a parent child device, then set the find_child function on init.
-    -- This is persisted because initialize switch is only run once, but find_child function should be set
-    -- on each driver init.
     device:set_field(IS_PARENT_CHILD_DEVICE, true, {persist = true})
   end
 
   device:set_field(SWITCH_INITIALIZED, true)
 
-  if component_map_used then
-    device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
-  end
+  -- this is needed in initialize_buttons_and_switches
+  return num_switch_server_eps
+end
 
-  if #button_eps > 0 and is_supported_combination_button_switch_device_type(device, main_endpoint) then
-    if #button_eps == 1 then
-      profile_name = "light-level-button"
-    else
-      profile_name = "light-level" .. string.format("-%d-button", #button_eps)
+local function handle_light_switch_with_onOff_server_clusters(device, main_endpoint, num_switch_server_eps)
+    local cluster_id = 0
+    for _, ep in ipairs(device.endpoints) do
+      -- main_endpoint only supports server cluster by definition of get_endpoints()
+      if main_endpoint == ep.endpoint_id then
+        for _, dt in ipairs(ep.device_types) do
+          -- no device type that is not in the switch subset should be considered.
+          if (ON_OFF_SWITCH_ID <= dt.device_type_id and dt.device_type_id <= ON_OFF_COLOR_DIMMER_SWITCH_ID) then
+            cluster_id = math.max(cluster_id, dt.device_type_id)
+          end
+        end
+        break
+      end
     end
-    device:try_update_metadata({profile = profile_name})
-    device:set_field(DEFERRED_CONFIGURE, true)
-    device:set_field(BUTTON_DEVICE_PROFILED, true)
-  elseif #button_eps > 0 then
-    local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
-    local battery_support = false
-    if device.manufacturer_info.vendor_id ~= HUE_MANUFACTURER_ID and #battery_feature_eps > 0 then
-      battery_support = true
+
+    if device_type_profile_map[cluster_id] then
+      device:try_update_metadata({profile = device_type_profile_map[cluster_id]})
     end
+end
 
-    if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
-      if battery_support then
-        local attribute_list_read = im.InteractionRequest(im.InteractionRequest.RequestType.READ, {})
-        attribute_list_read:merge(clusters.PowerSource.attributes.AttributeList:read())
-        device:send(attribute_list_read)
-      else
-        if #temperature_eps > 0 and #humidity_eps > 0 then
-          -- for now, this logic only applies to the Aqara Climate Sensor W100.
-          profile_name = "-temperature-humidity"
-        end
+local function initialize_buttons_and_switches(driver, device, main_endpoint)
```

Comment:
nit: A handful of the new helper functions used here use the `try_build_<thing>` format. However, there is no return or any indication if the attempt was successful. For clarity, I would just remove the `try` part unless we plan to handle failure cases (which based on previous functionality, I would assume we don't because they are unlikely)